# MANOVA 
## Author :  Muhammad Ali Saagar

**Python Developer • Data Science • AI/ML Engineer**

### 🔗 Connect With Me

[![GitHub](https://img.shields.io/badge/GitHub-Profile-181717?style=for-the-badge\&logo=github)](https://github.com/AliMuhammad78)
[![LinkedIn](https://img.shields.io/badge/LinkedIn-Profile-0A66C2?style=for-the-badge\&logo=linkedin)](https://www.linkedin.com/in/muhammad-ali-91294a290)

[![Kaggle](https://img.shields.io/badge/Kaggle-Profile-20BEFF?style=for-the-badge\&logo=kaggle)](https://www.kaggle.com/ali98muhammad45)
[![YouTube](https://img.shields.io/badge/YouTube-Channel-FF0000?style=for-the-badge\&logo=youtube)](https://www.youtube.com/@TechyBits20121)

[![Instagram](https://img.shields.io/badge/Instagram-Profile-E4405F?style=for-the-badge\&logo=instagram)](https://www.instagram.com/alisaagar6453/)
[![Facebook](https://img.shields.io/badge/Facebook-Profile-1877F2?style=for-the-badge\&logo=facebook)](https://www.facebook.com/profile.php?id=100089566913912)

[![Email](https://img.shields.io/badge/Email-Contact-D14836?style=for-the-badge\&logo=gmail)](mailto:alisaagar86@gmail.com)

---



## Acknowledgement

> Most of the Plotly concepts and examples in this notebook are based on the excellent tutorials by **Dr. Muhammad Aammar Tufail (Codanics)**. I sincerely thank him for sharing his knowledge and practical examples.

**Profiles:**
- GitHub: https://github.com/AammarTufail
- YouTube: https://www.youtube.com/@codanics
- Kaggle: https://www.kaggle.com/muhammadaammartufail
- LinkedIn: https://www.linkedin.com/in/dr-muhammad-aammar-tufail-02471213b/

---

 
## MANOVA

Manova is a multivariate extension of ANOVA. It is used to model two or more dependent variables that are continuous with one or more categorical predictor variables. It is often used to assess for differences between two or more groups.

To perform a Multivariate Analysis of Variance (MANOVA) in Python, we typically use the `statsmodels` library. MANOVA is used when there are two or more dependent variables and one or more independent variables. It tests whether the mean differences among groups on a combination of dependent variables are likely to have occurred by chance.

Here's an example demonstrating how to create a MANOVA table in Python:

### Example: MANOVA with StatsModels

Let's say we have a dataset with two dependent variables (e.g., test scores in mathematics and science) and one independent variable (e.g., teaching method). We want to know if there are statistically significant differences in the dependent variables across the levels of the independent variable.

### Explanation:

- **Dataset Preparation**: The `data` dictionary and `DataFrame` (`df`) contain the sample data. Replace this with your actual data.
- **MANOVA Execution**: The `MANOVA.from_formula` method is used to perform the MANOVA. The formula 'MathScore + ScienceScore ~ Method' indicates that `MathScore` and `ScienceScore` are dependent variables, and `Method` is the independent variable.
- **Results**: The `mv_test()` method is used to get the MANOVA test results, which are printed to the console.

This script will output the MANOVA table, including Pillai's trace, Wilks' lambda, Hotelling-Lawley trace, and Roy's greatest root test statistics, along with their associated F-values, degrees of freedom, and p-values. These results will help you determine if there are statistically significant differences in the dependent variables across the levels of the independent variable.

In [2]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.multivariate.manova import MANOVA

# Example dataset
data = {
    'Method': ['A', 'A', 'A', 'B', 'B', 'B', 'C', 'C', 'C'],
    'MathScore': [25, 28, 24, 31, 29, 32, 27, 26, 30],
    'ScienceScore': [34, 36, 33, 39, 37, 40, 35, 32, 38]
}

df = pd.DataFrame(data)
print(df.head(10))
# Perform MANOVA
maov = MANOVA.from_formula('MathScore + ScienceScore ~ Method', data=df)
print(maov.mv_test())

  Method  MathScore  ScienceScore
0      A         25            34
1      A         28            36
2      A         24            33
3      B         31            39
4      B         29            37
5      B         32            40
6      C         27            35
7      C         26            32
8      C         30            38
                  Multivariate linear model
                                                              
--------------------------------------------------------------
       Intercept         Value   Num DF Den DF F Value  Pr > F
--------------------------------------------------------------
          Wilks' lambda   0.0070 2.0000 5.0000 352.8955 0.0000
         Pillai's trace   0.9930 2.0000 5.0000 352.8955 0.0000
 Hotelling-Lawley trace 141.1582 2.0000 5.0000 352.8955 0.0000
    Roy's greatest root 141.1582 2.0000 5.0000 352.8955 0.0000
--------------------------------------------------------------
                                                 


## **Interpretation of MANOVA Results**

The MANOVA results test whether the three **Methods (A, B, C)** have significantly different combined mean scores for **MathScore and ScienceScore**.

### **Intercept Part**

The intercept results are statistically significant for all four test statistics:

1. **Wilks' Lambda:** 0.0070, with a p-value of 0.0000.
2. **Pillai's Trace:** 0.9930, with a p-value of 0.0000.
3. **Hotelling-Lawley Trace:** 141.1582, with a p-value of 0.0000.
4. **Roy's Greatest Root:** 141.1582, with a p-value of 0.0000.

These results indicate that the overall intercept is statistically significant. However, the **main focus of this MANOVA is the Method effect**, not the intercept.

### **Method Part**

The `Method` variable shows somewhat different results across the four multivariate test statistics:

1. **Wilks' Lambda:** 0.2061, F = 3.0071, p = 0.0719  
   → Not statistically significant at α = 0.05.

2. **Pillai's Trace:** 1.0370, F = 3.2308, p = 0.0512  
   → Very close to 0.05, but still not statistically significant at α = 0.05.

3. **Hotelling-Lawley Trace:** 2.6728, F = 3.2803, p = 0.1094  
   → Not statistically significant at α = 0.05.

4. **Roy's Greatest Root:** 2.1150, F = 6.3451, p = 0.0331  
   → Statistically significant at α = 0.05.

### **Overall Interpretation**

The four MANOVA tests do not give a completely consistent conclusion. **Wilks' Lambda, Pillai's Trace, and Hotelling-Lawley Trace have p-values greater than 0.05**, so they do not provide sufficient evidence that Method has a statistically significant multivariate effect on the combined Math and Science scores.

**Roy's Greatest Root** gives a significant result (p = 0.0331), suggesting a possible difference driven strongly by one multivariate dimension.

Because the commonly preferred MANOVA statistics, particularly **Pillai's Trace and Wilks' Lambda**, are not significant at the 5% level, we would generally conclude that there is **not strong enough overall evidence to reject the null hypothesis that the methods have the same combined mean outcomes**.

If the MANOVA is considered significant based on a chosen test statistic, follow-up **univariate ANOVAs and post-hoc comparisons** can be performed to determine which individual dependent variable and which groups are responsible for the observed differences.